apps/studio/data/table-rows/table-rows-query.ts

```diff
@@ -94,22 +102,26 @@ export const getTableRowsSqlQuery = ({
 }: GetTableRowsArgs) => {
   const query = new Query()
 
-  if (!table) {
-    return ``
-  }
+  if (!table) return ``
 
-  const enumArrayColumns = table.columns
+  // [Joshen] Only truncate text/json based columns as their length could go really big
+  // Note: Risk of payload being too large if the user has many many text/json based columns
+  // although possibly negligible risk.
+  const truncatedColumns = table.columns
     .filter((column) => {
-      return (column?.enum ?? []).length > 0 && column.dataType.toLowerCase() === 'array'
+      return TEXT_TYPES.includes(column.format) || JSON_TYPES.includes(column.format)
+    })
+    .map((column) => {
+      if ((column?.enum ?? []).length > 0 && column.dataType.toLowerCase() === 'array') {
+        return `"${column.name}"::text[]`
+      } else {
+        return `case when length("${column.name}"::text) > ${MAX_CHARACTERS} then concat(left("${column.name}"::text, ${MAX_CHARACTERS}), '...') else "${column.name}"::text end "${column.name}"`
+      }
     })
-    .map((column) => column.name)
 
-  let queryChains =
-    enumArrayColumns.length > 0
-      ? query
-          .from(table.name, table.schema ?? undefined)
-          .select(`*,${enumArrayColumns.map((x) => `"${x}"::text[]`).join(',')}`)
-      : query.from(table.name, table.schema ?? undefined).select()
+  let queryChains = query
+    .from(table.name, table.schema ?? undefined)
+    .select(`*,${truncatedColumns.join(',')}`)
```

Comment:
Might need to add a check here for when truncatedColumns is empty. Getting this:
<img width="736" alt="Screenshot 2024-05-27 at 20 54 04" src="https://github.com/supabase/supabase/assets/10985857/78f4331f-6d56-47ce-971e-90336da47b76">

Reply:
pushed up a fix! 🙏